JacksonDunstanNativeCollections/NativePerJobThreadIntPtrs.cs

```diff
@@ -0,0 +1,434 @@
+		/// This property should <i>not</i> be used to check whether the object
+		/// is usable, only to check whether it was <i>ever</i> usable.
+		/// </value>
+		public bool IsCreated
+		{
+			get
+			{
+				return m_Buffer != null;
+			}
+		}
+
+		/// <summary>
+		/// Release the object's unmanaged memory. Do not use it after this. Do
+		/// not call <see cref="Dispose"/> on copies of the object either.
+		/// 
+		/// This operation requires write access.
+		/// 
+		/// This complexity of this operation is O(1) plus the allocator's
+		/// deallocation complexity.
+		/// </summary>
+		[WriteAccessRequired]
+		public void Dispose()
+		{
+			RequireWriteAccess();
+
+// Make sure we're not double-disposing
+#if ENABLE_UNITY_COLLECTIONS_CHECKS
+#if UNITY_2018_3_OR_NEWER
+        	DisposeSentinel.Dispose(ref m_Safety, ref m_DisposeSentinel);
+#else
+			DisposeSentinel.Dispose(m_Safety, ref m_DisposeSentinel);
+#endif
+#endif
+
+			UnsafeUtility.Free(m_Buffer, m_AllocatorLabel);
+			m_Buffer = null;
+		}
+
+		/// <summary>
+		/// Set whether both read and write access should be allowed. This is
+		/// used for automated testing purposes only.
+		/// </summary>
+		/// 
+		/// <param name="allowReadOrWriteAccess">
+		/// If both read and write access should be allowed
+		/// </param>
+		[Conditional("ENABLE_UNITY_COLLECTIONS_CHECKS")]
+		[BurstDiscard]
+		public void TestUseOnlySetAllowReadAndWriteAccess(
+			bool allowReadOrWriteAccess)
+		{
+#if ENABLE_UNITY_COLLECTIONS_CHECKS
+			AtomicSafetyHandle.SetAllowReadOrWriteAccess(
+				m_Safety,
+				allowReadOrWriteAccess);
+#endif
+		}
+
+		/// <summary>
+		/// Throw an exception if the object isn't readable
+		/// </summary>
+		[Conditional("ENABLE_UNITY_COLLECTIONS_CHECKS")]
+		[BurstDiscard]
+		private void RequireReadAccess()
+		{
+#if ENABLE_UNITY_COLLECTIONS_CHECKS
+			AtomicSafetyHandle.CheckReadAndThrow(m_Safety);
+#endif
+		}
+
+		/// <summary>
+		/// Throw an exception if the object isn't writable
+		/// </summary>
+		[Conditional("ENABLE_UNITY_COLLECTIONS_CHECKS")]
+		[BurstDiscard]
+		private void RequireWriteAccess()
+		{
+#if ENABLE_UNITY_COLLECTIONS_CHECKS
+			AtomicSafetyHandle.CheckWriteAndThrow(m_Safety);
+#endif
+		}
+	}
+
+	/// <summary>
+	/// Provides a debugger view of <see cref="NativeIntPtr"/>.
+	/// </summary>
+	internal sealed class NativePerJobThreadIntPtrsDebugView
+	{
+		/// <summary>
+		/// The object to provide a debugger view for
+		/// </summary>
+		private NativePerJobThreadIntPtrs ptrs;
+
+		/// <summary>
+		/// Create the debugger view
+		/// </summary>
+		/// 
+		/// <param name="ptr">
+		/// The object to provide a debugger view for
+		/// </param>
+		public NativePerJobThreadIntPtrsDebugView(NativePerJobThreadIntPtrs ptrs)
+		{
+			this.ptrs = ptrs;
+		}
+
+		/// <summary>
+		/// Get the viewed object's value
+		/// </summary>
+		/// 
+		/// <value>
+		/// The viewed object's value
+		/// </value>
+		public int this[int index]
```

Comment:
The debugger won't call this indexer with `index` parameters. Instead, use an `Items` property that returns a managed array of values as in the other collections.